macros/src/lib.rs

```diff
@@ -836,27 +855,34 @@ fn check_attr_whitelist(attrs: &[Attribute]) -> Result<(), TokenStream> {
 
         for val in cortex_m_rt_blacklist {
             if eq(&attr, &val) {
-                return Err(
-                    parse::Error::new(
-                        attr.span(),
-                        "multiple cortex-m-rt attributes are not supported on the same function",
-                    )
+                let err_str = match caller {
+                    WhiteListCaller::Entry => {
+                        &"this attribute is not allowed on a cortex-m-rt entry point"
+                    }
+                    WhiteListCaller::Exception => {
+                        &"this attribute is not allowed on an exception handler"
+                    }
+                    WhiteListCaller::Interrupt => {
+                        &"this attribute is not allowed on an interrupt handler"
+                    }
+                    WhiteListCaller::PreInit => {
+                        &"this attribute is not allowed on an interrupt handler"
+                    }
+                };
+
+                return Err(parse::Error::new(attr.span(), err_str)
```

Comment:
Can do `&err_str` here and remove the `&`s in front of the string literals above